elements/lisk-validator/src/validation.ts

```diff
@@ -236,24 +214,30 @@ export const validateNetworkIdentifier = (
 	return true;
 };
 
-export const isInt32 = (num: bigint | number): boolean => {
-	if (typeof num === 'number') {
-		return num <= MAX_INT32 && num >= MIN_INT32;
+export const isString = (data: unknown): boolean => typeof data === 'string';
+
+export const isBoolean = (data: unknown): boolean => typeof data === 'boolean';
+
+export const isSInt32 = (data: unknown): boolean => {
+	if (typeof data === 'number' && Number.isInteger(data)) {
+		return data <= MAX_INT32 && data >= MIN_INT32;
 	}
 
-	return num <= BigInt(MAX_INT32) && num >= BigInt(MIN_INT32);
+	return false;
 };
```

Comment:
As now its `unknown`, should not we consider logic for `string` as well?

Reply:
you mean if data is `string`? that is being checked above `typeof data === 'number'`

Reply:
Yes I meant if number can be in string format.